api/providergen/converters.go

```diff
@@ -0,0 +1,91 @@
+package main
+
+import "text/template"
+
+var (
+	// ProviderConvertersTemplate is the template of the Terraform provider implementation for the Illumio CloudSecure Config API.
+	ProviderConvertersTemplate = template.Must(template.New("providermodel").Parse(`
+{{- define "convertersForModel"}}
+type {{.Name}} struct {
+	{{- range $field := .Fields}}
+	{{$field.Name}} types.{{$field.Type.ModelTypeName}} ` + "`" + `tfsdk:"{{$field.AttributeName}}"` + "`" + `
+	{{- end}}
+}
+
+func GetTypeAttrsFor{{.Name}}() map[string]attr.Type {
+	return map[string]attr.Type{
+		{{- range $field := .Fields}}
+		{{- if ne $field.Type.NestedModel nil}}
+		"{{$field.AttributeName}}": types.ObjectType{
+			AttrTypes: GetTypeAttrsFor{{$field.Type.NestedModel.Name}}(),
+		},
+		{{- else}}
+		"{{$field.AttributeName}}": types.{{$field.Type.ModelTypeName}}Type,
+		{{- end}}
+		{{- end}}
+	}
+}
+
+func Convert{{.Name}}ToObjectValueFromProto(proto *configv1.{{.Name}}) basetypes.ObjectValue  {
+	return types.ObjectValueMust(
+		GetTypeAttrsFor{{.Name}}(),
+		map[string]attr.Value{
+			{{- range $field := .Fields}}
+			{{- if ne $field.Type.NestedModel nil}}
+			"{{$field.AttributeName}}": Convert{{$field.Type.NestedModel.Name}}ToObjectValueFromProto(proto.{{$field.Name}}),
+			{{- else}}
+			"{{$field.AttributeName}}": types.{{$field.Type.ModelTypeName}}Value(proto.{{$field.Name}}),
+			{{- end}}
+			{{- end}}
+		},
+	)
+}
+
+func ConvertDataValueTo{{.Name}}Proto(dataValue attr.Value) (*configv1.{{.Name}}, diag.Diagnostics) {
+	pv := {{.Name}}{}
+	diags := tfsdk.ValueAs(context.Background(), dataValue, &pv)
```

Comment:
Pass in and use a proper Context.
I already commented on this in https://github.com/illumio/terraform-provider-illumio-cloudsecure/pull/107#discussion_r1882802313.